File: Easy/Python/747. Largest Number At Least Twice of Others.py
class Solution(object):
    def dominantIndex(self, nums):
        largest = 0        # Index of largest element
        secondLargest = 1  # Index of second largest element

        for i in range(1, len(nums)):
            if nums[largest] < nums[i]:
                secondLargest = largest
                largest = i
            elif nums[secondLargest] < nums[i]:
                secondLargest = i
        
        return largest if nums[secondLargest] * 2 <= nums[largest] else -1

File: Easy/JavaScript/747. Largest Number At Least Twice of Others.js
var dominantIndex = function(nums) {
    let largest = 0;        // Index of largest element
    let secondLargest = 1;  // Index of second largest element

    for(let i = 1; i < nums.length; i++){
        if(nums[largest] < nums[i]){
            secondLargest = largest;
            largest = i;
        }else if(nums[secondLargest] < nums[i])
            secondLargest = i;
    }

    return nums[secondLargest] * 2 <= nums[largest] ? largest : -1;
};
